torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -199,11 +199,20 @@ template class UserRRef<py::object>;
 
 //////////////////////////  OwnerRRef  /////////////////////////////////////
 
-template <typename T>
-const T& OwnerRRef<T>::getValue() const {
+template <>
+const IValue& OwnerRRef<IValue>::getValue() const {
```

Comment:
Why do we have to separate IValue and python objects like this? Shouldn't the same code work in python/torchscript mode since this is C++ code?

Reply:
because python object and IValue exception handing is different per previous request. 

For IValue exception handling, it catches C++ exception and returns a message with message::Exception, then exception will be rethrown based on message::Exception type. 

For pyobject exception handing, it did not throw a C++ exception, but catch it in python side and wrap it into a RemoteException to caller. Caller will check this returned result is RemoteException or not and then rethrown. This is based on previous request that we need to return python call stack trace to users. C++ exception can only return a message, no stack trace